E2E test for case "Save a draft of menu changes and preview them via …

igladun opened the pull request:
…custom URL"#787

hoverduck commented on specs/wp-post-nux-spec.js:
```diff
@@ -66,18 +66,6 @@ test.describe( `[${host}] Post-NUX Flows (${screenSize}) @parallel`, function()
 					} );
 				} );
 
-				test.it( 'Can update and view site tagline', function() {
```
Is there a reason you took this out?

alisterscott commented on lib/base-container.js:
```diff
@@ -39,6 +39,12 @@ export default class BaseContainer {
 		return driverHelper.waitTillPresentAndDisplayed( this.driver, this.expectedElementSelector, this.explicitWaitMS );
 	}
 
+	waitRequestsToComplete() {
```
this will only work on the customizer itself which is the only place where jQuery will be available, so it would be best to move this into the `lib/pages/customizer-page.js` to make sure it's not called from other pages.

alisterscott commented on lib/base-container.js:
```diff
@@ -39,6 +39,12 @@ export default class BaseContainer {
 
+	waitRequestsToComplete() {
+		return this.driver.wait( () => {
+			return this.driver.executeScript( 'return jQuery.active === 0' );
```
This method should also check to make sure that jQuery is loaded also because sometimes switching iFrames there are delays (which is why we have the sleeps to begin with)

alisterscott commented on lib/base-container.js:
```diff
@@ -39,6 +39,12 @@ export default class BaseContainer {
+	waitRequestsToComplete() {
+		return this.driver.wait( () => {
+			return this.driver.executeScript( 'return jQuery.active === 0' );
+		}, 10000, 'Timed out waiting while AJAX loads' );
```
Where did this number come from? How did you decide upon it?
It should be declared at least as a constant and preferably as a config value. You could probably use `this.explicitWaitMs` which I believe has the same intention

igladun commented:
Thanks for the review. I fixed all remarks. The test was failing sometimes on mobile because the theme of test account is changing. Now should be OK.
@hoverduck I moved the test to separate file, because I think we will have more tests for the Customizer and to don't load the  `wp-post-nux`.

alisterscott commented:
Thanks for the updates @igladun 

I think this looks good now - my only concern is whether these menus will build up over time - what do you think @hoverduck?

hoverduck commented:
> my only concern is whether these menus will build up over time 

Yeah, now that you mention it I worry about that too.  I wasn't paying attention to the fact that that's a persistent change beyond the scope of the Customizer.  We could update the cleanup cron to also wipe out the menu, but it's probably better to have this test just modify a setting like the site title, or add a widget or something that only changes the front-end.

igladun commented:
After discussion with @hoverduck , closing this PR and creating a new one
